crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +137,122 @@ impl Persistence {
+                    .and_then(|p| domain::auction::Price::new(p.into()))
+                    .map_err(error::Auction::Price)?;
+                prices.insert(token, price);
+            }
+            prices
+        };
+
+        let fee_policies = {
+            let orders = database::auction_orders::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction orders")?
+                .ok_or(error::Auction::Missing)?
+                .into_iter()
+                .map(|order| domain::OrderUid(order.0));
+
+            let mut fee_policies = HashMap::new();
+            for order in orders {
+                fee_policies.insert(order, vec![]);
+                if database::orders::read_order(&mut ex, &ByteArray(order.0))
+                    .await
+                    .context("fetch order")?
+                    .is_some()
+                {
+                    let quote = database::orders::read_quote(&mut ex, &ByteArray(order.0))
+                        .await
+                        .context("fetch quote")?
+                        .map(dto::quote::into_domain)
+                        .transpose()
+                        .map_err(|_| error::Auction::DbConversion("quote overflow"))?
+                        .ok_or(error::Auction::MissingQuote)?;
+
+                    let policies =
+                        database::fee_policies::fetch(&mut ex, auction_id, ByteArray(order.0))
+                            .await
+                            .context("fetch fee policies")?;
+
+                    for policy in policies {
+                        let policy = dto::fee_policy::into_domain(policy, &quote)?;
+                        fee_policies.entry(order).and_modify(|policies| {
+                            policies.push(policy);
+                        });
+                    }
+                }
+            }
+            fee_policies
+        };
+
+        let surplus_capturing_jit_order_owners =
+            database::surplus_capturing_jit_order_owners::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch surplus capturing jit order owners")?
+                .ok_or(error::Auction::MissingJitOrderOwners)?
+                .into_iter()
+                .map(|owner| eth::H160(owner.0).into())
+                .collect();
+
+        Ok(domain::settlement::Auction {
+            id: auction_id,
+            prices,
+            fee_policies,
+            deadline,
+            surplus_capturing_jit_order_owners,
+        })
+    }
 }
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error("failed to read data from database")]
     DbError(#[from] anyhow::Error),
-    #[error(transparent)]
-    Price(#[from] domain::auction::InvalidPrice),
+}
+
+pub mod error {
+    use super::*;
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Auction {
+        #[error("failed to read data from database")]
+        DbError(#[from] anyhow::Error),
```

Comment:
This blanket conversion seems dangerous (it's inviting to simply handle any error with ? which would be translated into a DB error even if it's unrelated to DB operations). Can you please create the DbError explicitly in the cases where it makes sense (ie. remove the `#[from]` here)?

Reply:
removed